streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -476,7 +485,23 @@ public StreamsConfig(final Map<?, ?> props) {
 
         return consumerProps;
     }
-
+                 
+    private void checkIfUnexpectedUserSpecifiedConsumerConfig(Map clientProvidedProps) {
+        // Streams do not allow users to configure certain consumer configurations, for example,
+        // enable_auto_commit. In cases where user tries to override such non-configurable
+        // consumer configurations, log a warning and remove the user defined value from the Map.
+        // Thus the default values for these consumer configurations that are suitable for
+        // streams will be used instead.
+        for (int index = 0; index < NON_CONFIGURABLE_CONSUMER_CONFIGS.length; index++) {
```

Comment:
I used an enhanced for loop to iterate through a String array and also added final.